pkg/sensors/exec/exec_linux.go

```diff
@@ -139,41 +139,134 @@ func execParse(reader *bytes.Reader) (processapi.MsgProcess, bool, error) {
 		if n != -1 {
 			proc.Filename = strutils.UTF8FromBPFBytes(args[:n])
 			args = args[n+1:]
+		} else {
+			// Filename not null-terminated or buffer consumed
+			proc.Filename = strutils.UTF8FromBPFBytes(args[:])
+			args = nil
 		}
+	} else if exec.Flags&api.EventErrorFilename != 0 {
+		// Filename read error - buffer content is unreliable
+		args = nil
 	}
 
-	var cmdArgs [][]byte
-
-	if exec.Flags&api.EventDataArgs != 0 {
-		var desc dataapi.DataEventDesc
+	// At this point, 'args' holds the buffer containing [arguments_string]\0\0[environment_string]
+	// or 'args' is nil if filename parsing consumed everything or an error occurred.
+
+	var argsPayload []byte
+	var envsPayload []byte
+
+	// --- Robust args/envs split logic ---
```

Comment:
this change is too big and scary on very crucial code

if you need to change current args retrieval to fit in the environment stuff,
please do that in multiple incremental changes, so we can be sure we don't break things